models/applicantModel.js

```diff
@@ -1,28 +1,461 @@
+            route.router_index,
+            route.plane_needed,
+            route.hotel_needed,
+            route.beginning_date,
+            route.beginning_time,
+            route.ending_date,
+            route.ending_time,
+          ]);
+
+          const routeId = routeTableResult.insertId;
+
+          // ======================================
+          // Step 3: Insert into Route_Request table
+          // ======================================
+
+          const insertIntoRouteRequestTable = `
+          INSERT INTO route_request (request_id, route_id) VALUES (?, ?)
+        `;
+          await conn.query(insertIntoRouteRequestTable, [requestId, routeId]);
+        } catch (error) {
+          console.error("Error processing route:", error);
+          throw new Error("Database Error: Unable to process route");
+        }
+      }
+
+      await conn.commit();
+
+      console.log(`Travel request created with ID: ${requestId}`);
+      return {
+        requestId: Number(requestId),
+        message: "Travel request successfully created",
+      };
+    } catch (error) {
+      if (conn) await conn.rollback();
+      console.error("Error creating travel request:", error);
+      throw new Error("Database Error: Unable to fill Request table");
+    } finally {
+      if (conn) conn.release();
+    }
+  },
+
+  async editTravelRequest(requestId, travelChanges) {
+    let conn;
+    try {
+      conn = await pool.getConnection();
+      await conn.beginTransaction();
+      console.log("Editing travel request with ID:", requestId);
+
+      // Destructure travel details from request body
+      const {
+        router_index,
+        notes,
+        requested_fee = 0,
+        imposed_fee = 0,
+        origin_country_name,
+        origin_city_name,
+        destination_country_name,
+        destination_city_name,
+        beginning_date,
+        beginning_time,
+        ending_date,
+        ending_time,
+        plane_needed,
+        hotel_needed,
+        additionalRoutes = [],
+      } = travelChanges;
+
+      // Format the routes into a single array
+      const allRoutes = formatRoutes(
+        {
+          router_index,
+          origin_country_name,
+          origin_city_name,
+          destination_country_name,
+          destination_city_name,
+          beginning_date,
+          beginning_time,
+          ending_date,
+          ending_time,
+          plane_needed,
+          hotel_needed,
+        },
+        additionalRoutes
+      );
+
+      // =======================================
+      // Step 1: Update Request table
+      // =======================================
+      const request_days = getRequestDays(allRoutes);
+
+      // Log old data
+      const [oldData] = await conn.query(
+        `SELECT * FROM request WHERE request_id = ?`,
+        [requestId]
+      );
+      console.log("Old data:", oldData);
+
+      const updateRequestTable = `
+        UPDATE request SET
```

Comment:
```suggestion
        UPDATE Request SET
```
Incorrect name of the table